ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java

```diff
@@ -155,12 +156,30 @@ public void update(float tpf) {
             updateEnemies(tpf);
             checkObstacleCollision();
             checkPathCollision();
+            checkGameOver();
             updateTestWorld();
             hudController.updateHUD();
         }
     }
 
 
+    private void checkGameOver() {
+        int count = 0;
+        for (Carrier carrier: enemySpawner.getCarriers()) {
+            if (carrier.getHealth() <= 0) {
+                count++;
+            }
```

Comment:
Instead of this, make a default boolean isDead() method in the IKillable interface that returns true iff getHealth() <= 0.